explainer.md

```diff
@@ -186,225 +181,236 @@ https://user-images.githubusercontent.com/93594/141100217-ba1fa157-cd79-4a9d-b3b
+The other modes mentioned in this document, such as "computed style + content image", and "retaining hierarchy" will be exposed via other CSS properties, and `page-transition` will be used as a shorthand. However, this is out of scope for 'v1'.
 
-Using CSS properties means values can change depending on viewport (`@media`) and browser support (`@supports`).
+### Via JavaScript
 
-### JS API
+The JavaScript API extends on the capabilities of the CSS API. In particular, it allows the developer to react to the URL of Page-B.
 
 ```js
-// In Page-A
-addEventListener('navigate', (event) => {
-  event.prepareSameOriginPageTransition((transition) => {
-    transition.offerElement('header', document.querySelector('.header'), {
-      capture: 'container-and-child',
-    });
-
-    transition.setData({…});
-  });
+document.addEventListener("pagehide", (event) => {
+  if (!event.transition) return;
+  event.transition.setElement(document.documentElement, "root");
+  event.transition.setData(data);
 });
 ```
 
-Expanding on the capabilities of the CSS properties, a JS API allows the developer to change which parts are offered depending on the destination of the navigation, and the direction of navigation (back vs forward). This builds on top of the [app-history API](https://github.com/WICG/app-history).
+The [`pagehide`](https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event) event is an existing event which fires during the unload of a document. The event type is currently [`PageTransitionEvent`](https://html.spec.whatwg.org/multipage/indices.html#events-2:pagetransitionevent). This proposal will change that to `PageHideTransitionEvent`, which extends `PageTransitionEvent`, and adds the following:
 
-Also, data can be provided via `setData`. This can be anything structured-clonable, and will be made available to Page-B.
+- `event.nextURL` - The URL of the next page. Is `null` if cross-origin, or if there's no next page for this window.
+- `event.transition` - `null` if a transition cannot be performed. The following needs to be true for transitions to be performed:
+  - The browser supports page transitions.
+  - This document is navigating to another (`pagehide` also fires if the iframe/tab is being removed).
+  - The next page is same-origin, although cross-origin may be supported in future.
+  - The window is top-level, although nested page transitions may be supported in future.
+- `event.transition.setElement(element, tag, options)` - Set an element to be used in the transition.
+  - `element` - the element.
+  - `tag` - the tag name. Can be z to un-set this element.
+  - `options` - reserved for future use. This is where "computed style + content image" and "retaining hierarchy" modes will be exposed.
+- `event.transition.setData(data)` - An object that is structured-cloned and passed to the next page.
```

Comment:
I'm not going to add this, as it might suggest that other things may be supported for cross-origin cases, meaning we'll have to cover the doc in qualifications like "maybe not for cross-origin!"

The doc already contains:

> Cross-origin transitions are something we want to tackle, but may have significant differences and restrictions for security reasons. Cross-origin transitions are not covered in this document.

…at the start, and:

> Cross-origin transitions aren't yet defined, but are likely to be heavily restricted.

…at the end.

I'll add some other statements to make it clear that a transition will not happen cross-origin.